lib/Serialization/Deserialization.cpp

```diff
@@ -3446,8 +3446,10 @@ Type ModuleFile::getType(TypeID TID) {
 
   case decls_block::PAREN_TYPE: {
     TypeID underlyingID;
-    decls_block::ParenTypeLayout::readRecord(scratch, underlyingID);
-    typeOrOffset = ParenType::get(ctx, getType(underlyingID));
+    uint8_t flagsValue;
```

Comment:
Yep, please don't do this. Encode the flags individually instead and rebuild ParameterTypeFlags.